testData.js

```diff
@@ -59,17 +63,24 @@ class TestData {
         if (testURL) {
             var response = await HTTP.getJSON(testURL);
             return new TestData(response, query);
-        }
-        else {
-            throw new Error("A 'url' parameter should be included on the query string.");
+        } else if (query.path) {
+            const content = await readFile(path.join(config.testsDir, query.path));

```

Comment:
don't you want to make sure the path exists here as well?

Reply:
Thanks, done.